webserver/__init__.py

```diff
@@ -46,6 +51,17 @@ def create_app():
     from webserver.errors import init_error_handlers
     init_error_handlers(app)
 
+    # Add rate limit headers to responses
+    @app.after_request
+    def inject_x_rate_headers(response):
```

Comment:
I'd make this function more generic and move it into `rate_limiter` module. This way `create_app` will be somewhat cleaner and we can add new after-request tasks easily.

Something like this:
```python
from webserver import rate_limiter

@app.after_request
def after_request_callbacks(response):
    return rate_limiter.inject_headers(response)
```